src/util/json.cpp

```diff
@@ -162,3 +164,48 @@ void jsont::swap(jsont &other)
   other.object.swap(object);
   other.value.swap(value);
 }
+
+bool operator==(const jsont &left, const jsont &right)
+{
+  if(left.kind != right.kind)
+    return false;
+  switch(left.kind)
+  {
+  case jsont::kindt::J_NULL:
+    return true;
+  case jsont::kindt::J_TRUE:
+    return true;
+  case jsont::kindt::J_FALSE:
+    return true;
+  case jsont::kindt::J_NUMBER:
+    return left.value == right.value;
+  case jsont::kindt::J_STRING:
+    return left.value == right.value;
+  case jsont::kindt::J_ARRAY:
+  {
+    const auto &left_array = static_cast<const json_arrayt &>(left);
+    const auto &right_array = static_cast<const json_arrayt &>(right);
+    return left_array.size() == right_array.size() &&
+           std::equal(
+             left_array.begin(), left_array.end(), right_array.begin());
+  }
+  case jsont::kindt::J_OBJECT:
+  {
+    const auto &left_object = static_cast<const json_objectt &>(left);
+    const auto &right_object = static_cast<const json_objectt &>(right);
+    const size_t left_size =
+      narrow<size_t>(std::distance(left_object.begin(), left_object.end()));
+    const size_t right_size =
+      narrow<size_t>(std::distance(right_object.begin(), right_object.end()));
```

Comment:
Why do those conversions when afterwards you're only going to use the values for comparing them to each other? I'd go with `const auto left_size = std::distance ...`